gnat2goto/driver/tree_walk.adb

```diff
@@ -1639,12 +1727,20 @@ package body Tree_Walk is
                   Cond : Irep;
                   Post : Irep;
 
-                  Bound_Low  : constant Irep :=
-                    Do_Expression (Low_Bound (Dsd));
-                  Bound_High : constant Irep :=
-                    Do_Expression (High_Bound (Dsd));
-
+                  Bound_Low  : Irep;
+                  Bound_High : Irep;
                begin
+                  --  Produce something on which we can call Low_Bound and
+                  --  High_Bound
+                  if Nkind (Dsd) = N_Subtype_Indication then
```

Comment:
Usual pattern for this: make a little function to extract the Dsd, then let the main function have the same structure as before, but call the helper (e.g. `Dsd : constant Node_Id := Get_Dsd (Spec); Bound_Low  : constant Irep := Do_Expression (Low_Bound (Dsd)); ...`)